### PR TITLE
Add selector to pick targets based on calibration role

### PR DESCRIPTION
**Pull request by cquiroz**

### CODE REVIEW

File: modules/service/src/main/scala/lucuma/odb/graphql/input/WhereOptionEq.scala [hunk 1]
```diff
@@ -25,8 +26,8 @@ object WhereOptionEq {
           (isNull, EQ, NEQ, IN, NIN) =>
             and(List(
               isNull.map(IsNull(path, _)),
-              EQ.map(a => Eql(path, Const(a))),
-              NEQ.map(a => NEql(path, Const(a))),
+              EQ.map(a => Eql(path, Const(a.some))),
+              NEQ.map(a => NEql(path, Const(a.some))),
               IN.map(as => In(path, as)),
```
Comment (swalker2m) [inline modules/service/src/main/scala/lucuma/odb/graphql/input/WhereOptionEq.scala:31]: Should that be `as.map(_.some)` for `IN` and `NIN`?  Maybe add a test or change the `EQ` test to `IN`?

File: modules/service/src/test/scala/lucuma/odb/graphql/DatabaseOperations.scala [hunk 2]
```diff
@@ -1264,6 +1266,24 @@ trait DatabaseOperations { this: OdbSuite =>
...
+        c_library_desc
+      )
+      select
+        $calibration_role,
+        $text
+      returning c_program_id
+    """.apply(calibrationRole, description.value)
+    FMain.databasePoolResource[IO](databaseConfig).flatten
+      .use(_.prepareR(af.fragment.query(program_id)).use(_.unique(af.argument)))
+  }
```
Comment (swalker2m) [inline modules/service/src/test/scala/lucuma/odb/graphql/DatabaseOperations.scala:1285]: There isn't a service method that would do this?

**Review by swalker2m — APPROVED** [review]
Looks good, with a couple of observations.

File: modules/service/src/main/scala/lucuma/odb/graphql/input/WhereOptionEq.scala (hunk 1 above)
Comment (cquiroz) [reply modules/service/src/main/scala/lucuma/odb/graphql/input/WhereOptionEq.scala:31]: I'll add a test for `IN`

File: modules/service/src/test/scala/lucuma/odb/graphql/DatabaseOperations.scala (hunk 2 above)
Comment (cquiroz) [reply modules/service/src/test/scala/lucuma/odb/graphql/DatabaseOperations.scala:1285]: Not really as it is not expected to create system programs programmatically. For example, it isn't allow in the API.

Comment (swalker2m) [reply modules/service/src/test/scala/lucuma/odb/graphql/DatabaseOperations.scala:1285]: Wouldn't the programmatic update use the service?  If we can, I think we want to funnel all these database manipulations into service code.

**Review by swalker2m — APPROVED** [review]
This looks great to me.  Thanks for fixing the bug I left in `IN` and `NIN`! :+1:

**Pull request merged**